src/objectid.ts

```diff
@@ -47,9 +39,12 @@ export class ObjectId {
   /** @internal */
   static index = ~~(Math.random() * 0xffffff);
 
-  static cacheHexString?: boolean;
-  id: string | Buffer;
-  __id?: string;
+  static cacheHexString: boolean;
+
+  /** ObjectId Bytes @internal */
+  private _id: Buffer;
```

Comment:
let's use a symbol here if we're going to take the approach of providing a getter/setter for the bytes?